tests/test_transcription.py

```diff
@@ -0,0 +1,201 @@
+                                           est_pitch, offset_ratio=None)
+
+    assert matching == [(0, 0), (1, 1), (3, 3)]
+
+
+def test_match_notes_strict():
+
+    ref_int, ref_pitch = np.array([[0, 1]]), np.array([100])
+    est_int, est_pitch = np.array([[0.05, 1]]), np.array([100])
+
+    matching = \
+        mir_eval.transcription.match_notes(ref_int, ref_pitch, est_int,
+                                           est_pitch, strict=True)
+
+    assert matching == []
+
+
+def test_precision_recall_f1():
+
+    # load test data
+    ref_int, ref_pitch = REF[:, :2], REF[:, 2]
+    est_int, est_pitch = EST[:, :2], EST[:, 2]
+
+    precision, recall, f_measure = \
+        mir_eval.transcription.precision_recall_f1(ref_int, ref_pitch, est_int,
+                                                   est_pitch)
+
+    scores_gen = np.array([precision, recall, f_measure])
+    scores_exp = np.array([SCORES['Precision'], SCORES['Recall'],
+                           SCORES['F-measure']])
+    assert np.allclose(scores_exp, scores_gen, atol=A_TOL)
+
+    precision, recall, f_measure = \
+        mir_eval.transcription.precision_recall_f1(ref_int, ref_pitch, est_int,
+                                                   est_pitch,
+                                                   offset_ratio=None)
+
+    scores_gen = np.array([precision, recall, f_measure])
+    scores_exp = np.array([SCORES['Precision_no_offset'],
+                           SCORES['Recall_no_offset'],
+                           SCORES['F-measure_no_offset']])
+    assert np.allclose(scores_exp, scores_gen, atol=A_TOL)
+
+
+def __check_score(score, expected_score):
+    assert np.allclose(score, expected_score, atol=A_TOL)
+
+
+def test_regression():
+
+    # Regression tests
+    ref_files = sorted(glob.glob(REF_GLOB))
+    est_files = sorted(glob.glob(EST_GLOB))
+    sco_files = sorted(glob.glob(SCORES_GLOB))
+
+    for ref_f, est_f, sco_f in zip(ref_files, est_files, sco_files):
+        with open(sco_f, 'r') as f:
+            expected_scores = json.load(f)
+        # Load in reference transcription
+        ref_int, ref_pitch = mir_eval.io.load_valued_intervals(ref_f)
+        # Load in estimated transcription
+        est_int, est_pitch = mir_eval.io.load_valued_intervals(est_f)
+        scores = mir_eval.transcription.evaluate(ref_int, ref_pitch, est_int,
+                                                 est_pitch)
+        for metric in scores:
+            # This is a simple hack to make nosetest's messages more useful
+            yield (__check_score, scores[metric], expected_scores[metric])
+
+
+def test_invalid_pitch():
+
+    ref_int, ref_pitch = np.array([[0, 1]]), np.array([-100])
+    est_int, est_pitch = np.array([[0, 1]]), np.array([100])
+
+    yield raises(ValueError)(mir_eval.transcription.validate), \
+        ref_int, ref_pitch, est_int, est_pitch
+    yield raises(ValueError)(mir_eval.transcription.validate), \
```

Comment:
Nitpick: https://docs.python.org/2/howto/doanddont.html#using-backslash-to-continue-statements